networkit/cpp/matching/SuitorMatcher.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ *  SuitorMatcher.cpp
+ *
+ *  Created on: 27.08.2019
+ *  Author: Michal Boron     <michal.s.boron@gmail.com>
+ *          Eugenio Angriman <angrimae@hu-berlin.de>
+ */
+
+// networkit-format
+
+#include <algorithm>
+#include <stdexcept>
+
+#include <networkit/matching/SuitorMatcher.hpp>
+
+namespace NetworKit {
+
+SuitorMatcher::SuitorMatcher(const Graph &G, bool sortSuitor, bool checkSortedEdges)
+    : Matcher(G), sortSuitor(sortSuitor) {
+
+    if (G.numberOfSelfLoops() > 0)
+        throw std::runtime_error("This algorithm does not graphs with self-loops.");
+
+    if (G.isDirected())
+        throw std::runtime_error("This algorithm does not support directed graphs.");
+
+    if (sortSuitor && checkSortedEdges && !hasEdgesSortedByWeight(G))
+        throw std::runtime_error("Edges are not sorted by weight");
+}
+
+bool SuitorMatcher::hasEdgesSortedByWeight(const Graph &G) {
+    for (node u : G.nodeRange()) {
+        node prevNode = none;
+        edgeweight prevWeight = std::numeric_limits<edgeweight>::max();
+        bool sorted = true;
+        G.forNeighborsOf(u, [&](node v, edgeweight ew) {
+            if (ew == prevWeight) {
+                if (prevNode > v)
+                    sorted = false;
+
+            } else if (ew > prevWeight)
+                sorted = false;
+
+            prevNode = v;
+            prevWeight = ew;
```

Comment:
This seems like a re-implementation of [`std::is_sorted`](https://en.cppreference.com/w/cpp/algorithm/is_sorted) on `G.weightNeighborRange` with a custom comparison function.